torch/utils/data/datapipes/iter/combining.py

```diff
@@ -96,7 +97,31 @@ def __new__(cls, datapipe: IterDataPipe, num_instances: int, buffer_size: int =
         return [_ChildDataPipe(container, i) for i in range(num_instances)]
 
 
-class _ForkerIterDataPipe(IterDataPipe):
+class _ContainerTemplate(ABC):
+    r"""
+    Abstract class for container ``DataPipes``. The followings are three required
+    methods.
+    """
+    @abstractmethod
+    def get_next_element_by_instance(self, instance_id: int):
+        ...
+
+    @abstractmethod
+    def is_every_instance_exhausted(self) -> bool:
+        ...
+
+    @abstractmethod
+    def reset(self) -> None:
+        ...
+
+    @abstractmethod
+    def get_length_by_instance(self, instance_id: int):
+        r"""
+        Raise TypeError if it's not supposed to be implemented to support `list(datapipe)`
+        """
```

Comment:
Add another method to add the functionality to provide different length per Child.
cc: @NivekT 
I need it for `demux` have a deterministic `classifier_fn`.

Reply:
https://github.com/pytorch/data/pull/903 is an example of the usage of `get_length_by_instance`.